### PR TITLE
Minimum Cost Of Ropes

### PR DESCRIPTION
**Pull request by utkarsh006**
## Related Issue
- Minimum Cost of ropes #1407

## Proposed Changes
- I have added the following things: -
- Problem Statement
- Examples
- Algorithm
- Code
- Time Complexity & Space Complexity

## Additional Info
[LINK OF THE PROBLEM STATEMENT](https://practice.geeksforgeeks.org/problems/minimum-cost-of-ropes-1587115620/1)

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

We were supposed to find the minimum cost of multiple ropes so that they can be connected with each other.

I have added the Problem Statement, Examples, an Algorithm on how to approach this problem, a commented Code, and written the Time Complexity and Space Complexity. And finally updated the README.md.

This change helps to solve a very famous problem of Heap i.e. Minimum Cost Of Ropes and this problem helps in understanding the concept of Heap.

I coded this problem in VS Code and tested the given problem on the test cases in the range: 1 ≤ N ≤ 100000 and 
1 ≤ arr[i] ≤ 10^6. And it is working fine for them.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
|Input : n= 4 and arr[]={4,3,2,6}| Input : n= 5 and arr[]={4,2,7,6,9}|
|image|image|

### CODE REVIEW

**Review by Mrjoy832 — APPROVED** [review]

**Comment by Mrjoy832** [comment]
LGTM🌟

**Comment by utkarsh006** [comment]
@Lakhankumawat Review my pr and kindly upgrade the level if you find it legible. My previous pr's were 
1385
1352
1337
1300

**Comment by Lakhankumawat** [comment]
Remove `Of` from file name

**Comment by utkarsh006** [comment]
@Lakhankumawat Kindly merge my Pr since I have done the requested changes. Also, upgrade the level of my Pr if you found it legible.

File: H-Heap/README.md [hunk 1]
```diff
@@ -71,3 +72,35 @@ Arr[] = {1, 2, 7, 3, 9, 19}
...
+
+
+# Minimum Cost Of Ropes
+- [Problem Statement](#problem-statement-1)
+- [Examples](#examples)
+- [Algorithm](#algorithm)
+- [Time Complexity and Space Complexity](#time-complexity-and-space-complexity)
+
+## Problem Statement 
+There are given N ropes of different lengths, we need to connect these ropes into one rope. The cost to connect two ropes is equal to sum of their lengths.
```
Comment (Lakhankumawat) [inline H-Heap/README.md:84]: No Image ?

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

**Comment by utkarsh006** [comment]
@Lakhankumawat I have added the image

**Review by Lakhankumawat — APPROVED** [review]
## LGTM ✨🎉🦖

**Comment by Lakhankumawat** [comment]
Next PR you'll get Level2 , i can't give level2 to you , although your contribution is good 

Keep Hustling 💎

**Pull request merged**